functionary/vllm_monkey_patch/async_llm_engine.py

```diff
@@ -237,6 +238,11 @@ async def step_async(self) -> List[RequestOutput]:
             prompt_template = self.prompt_templates[request_id]
             gen_state = self.gen_states[request_id]
             tools_or_functions = self.tools_or_functions[request_id]
+            # Check whether self.tokenizer is a TokenizerGroup class
+            if hasattr(self.tokenizer, "tokenizer"):
+                tokenizer = self.tokenizer.tokenizer
```

Comment:
Is this new in vllm 3.0 ?

Reply:
Yes, this is new in vllm 0.3.0, so I check it here and always pass HF AutoTokenizer into template class